### DISCUSSION TITLE
Using bazel7/bzlmod to build a connectrpc, go app does not generate the connect code, only protobuf code is created.

### DISCUSSION THREAD

**Discussion by abitofhelp** [connectrpc/connect-go#742] [Q&A]
Hello,

I have a simple application based on the ubiquitous Greeter example at grpc's website.  It uses go, connectrpc-go, and bazel7/bzlmod.  I can generate the connect and protobuf files using 'buf generate proto', build, and run the server and client. 

Now, I need to configure it to build using bazel7/bzlmod.  I can successfully generate protobuf but not connect proto libs.  I have scoured the Internet but have not found anyone trying to use the latest features in Bazel to build a connectrpc app, so I am hoping that we can sort out the issue so others can find it.  The repository where I am working is read-only, public, so you can see what I have created.

https://github.com/abitofhelp/bzlmod

I appreciate your time and interest!
Mike

**Comment by perezd** [comment]
Thanks for the report; that link isn't working...can you double-check it?

**Reply by abitofhelp** [reply]
My apologies...  I forgot to change the visibility to public.  You should be good to go.  Please let me know if you need anything from me.  I can adjust security so you can work with the repo, if desired.  Thanks for your help!

**Comment by emcfarlane** [comment]
Hey @abitofhelp, we currently don't have a predefined bazel rules for invoking `protoc-gen-connect-go`. To do this with rules_go see the support for declaring custom plugins here: https://github.com/bazelbuild/rules_go/blob/master/proto/core.rst#go_proto_compiler

**Reply by abitofhelp** [reply]
@emcfarlane Well, I am very, very close to having Bazel generate the PB and Connect files similarly to 'buf generate'.  

Source Code: https://github.com/abitofhelp/bzlmod.git
                        bazel branch
                        
                        
                        
The good news: 
```
bazel build //proto/helloworld/v1:helloworld_v1_go_proto
generates
bazel-bin/proto/helloworld/v1/helloworld_v1_go_proto_/github.com/abitofhelp/bzlmod/proto/helloworld/v1/helloworld.pb.go
bazel-bin/proto/helloworld/v1/helloworld_v1_go_proto_/github.com/abitofhelp/bzlmod/proto/helloworld/v1/helloworld.connect.pb.go
```

The bad news:
I am struggling to configure options for the Connect plugin to generate files like 'buf generate':

* Output the connect file to a subdirectory similar to what 'buf generate' creates.  At this time, the pb and connect files have different packages but exist in the same folder, which is a no-no.

* Sets the package name in the connect file to the subdirectory name so it is a proper Go module.  At this time, it is setting the name to 'v1connect' rather than 'helloworldv1connect', which 'buf generate' would create.

I am sure that these issues are simple configuration items for the go_proto_compiler block in the proto's BUILD.bazel file, but after a day of docs and experimenting, I haven't been able to resolve these final issues.  

Here is the go_proto_compiler block:
```
go_proto_compiler(
    name = "helloworld_v1_proto_connect_compiler",
    options = [
        "lang=go",
        "connect-go_out=./helloworldv1connect",
    ],
    plugin = "@com_connectrpc_connect//cmd/protoc-gen-connect-go:protoc-gen-connect-go",
    suffix = ".connect.go",
    valid_archive = False,
    visibility = ["//visibility:public"],
    deps = ["@com_connectrpc_connect//:go_default_library"]
)
```

Any help would be appreciated!

**Reply by abitofhelp** [reply]
The following protoc command properly generates the protobuf and connect files.  I am struggling to map it to the go_proto_compiler's configuration for the Connect plugin.  I'd appreciate your suggestions.

```
 protoc \
    -I proto \
    --go_out='internal/gen/' \
    --go_opt=paths=source_relative \
    --connect-go_out='internal/gen/' \
    --connect-go_opt=paths=source_relative \
    $(find proto -name '*.proto')
```
